### DISCUSSION TITLE
Open the cloud code as FOSS!

### DISCUSSION THREAD

**Discussion by avelino** [moclojer/moclojer#293] [General] [answered]
I know we've talked about this, and I was the one who hit the brakes saying not to open it, but I've been thinking a lot this week about the future of moclojer and the arrival of @J0sueTM at @buserbrasil _(main job)_.

I believe it makes sense to keep the project open with the goal of giving visibility to your work and using moclojer cloud as a use case for the packages we maintain open source.

@moclojer/company I would like to hear your opinion.

**Comment by Felipe-gsilva** [comment]
I really like the idea of opening code as open source. In any case, I can't see why not to do it !! 

I think the idea of moclojer is, but not only, to improve visibility and access to the nice clojure environment we are helping to build and using everyday. It feels nice and kind of the whole point of building great software to me :pray: 

What do you guys think are the cons of opening it?

**Comment by J0sueTM** [comment]
It has been a great pleasure for me to work on moclojer these past years, be it the open source libraries or moclojer-cloud. It culminates years of work from you @avelino, @matheusfrancisco, me and @Felipe-gsilva . I'm not an emotional person overall, but our small community has a great place in my heart, since it was my first opportunity to practice, learn, influence and drive a real project with my own ideas, and of course... code; but not only it. It was an opportunity for me to network with great people like you guys. I've learnt so much, and definitely have much more to learn. I'm really grateful for all we've been able to achieve so far.

Holding my tears for a little... I definitely approve the idea of open sourcing moclojer. It not only shows the world what we have been building so far, but also invites people who are interested in the idea, and can't wait for 2 busy young adults to finish the product :smile: , and do it themselves.

Besides that, I really like the community we have built so far. It's really small, but it does matter. I agree with you @avelino , we should continue maintaining our open source libraries. But alsoooo, why not continue building stuff? One thing I really liked about the way we do stuff is that we don't wait for things to fix themselves. A redis lib isn't working? Let's build our own. How many great libraries with great edge cases could we find by building stuff and fucking around?

I might have gone a bit overboard, but that's what have been going on in my mind. :eyes:

**Reply by avelino** [reply, marked as answer]
> One thing I really liked about the way we do stuff is that we don't wait for things to fix themselves. A redis lib isn't working?

This sentence sums up the open source world of open collaboration. It's not simply about something not working, so let's isolate ourselves in our _"little world"_ and do our thing. We started by trying to use what wasn't working for our use case, we tried to contribute and weren't _"heard"_ as we would have liked - I know people think differently and act completely differently, which is what most catches my attention in open source, the software is made by people to solve real problems, but each person will try to solve it in their own way, and it's okay for that to happen.

After attempting to contribute and feeling that we were moving further away from solving _"our problem"_ each day, we took the step of writing what worked for us, with an extremely clear objective and simple references (coming from another stack, aka Python RQ, Zen of Python: "simple is better than complex" `import this`).

---

Today we had an offsite with the engineering team from @buserbrasil _(literally today)_ and I bring to the company what I have lived and learned in the open source culture for some years:
- start simple and iterate until we reach where we want;
- small teams can accomplish great things;
- ... many more topics we talked about above

---

Happy that everyone agrees to transform what we were doing privately into open source!

> [!NOTE]
> **moclojer community:** get ready to receive another open source project, with many crazy ideas and code to use as a study base for your projects! We do everything under the MIT license and the moclojer cloud codebase will be no different 🎉
